src/libraries/System.Text.Json/tests/Serialization/NullableTests.cs

```diff
@@ -394,5 +394,47 @@ IEnumerator IEnumerable.GetEnumerator()
                 return ((IDictionary<string, TValue>)dict).GetEnumerator();
             }
         }
+
+        [Fact]
+        public static void NullableCustomStructRoundtrip()
+        {
+            string serialized = JsonSerializer.Serialize(new ClassWithNullablePerson
+            {
+                Person = new Person
+                {
+                    FirstName = "John",
+                    Age = 24
+                }
+            });
+            Assert.Contains(@"{""Person"":{", serialized);
+            Assert.Contains(@"""FirstName"":""John""", serialized);
+            Assert.Contains(@"""Age"":24", serialized);
+            Assert.Contains(@"""Birthday"":null", serialized);
+            Assert.DoesNotContain(@"""Value"":{""", serialized);
```

Comment:
Was there a previous issue with writing "Value" and "HasValue"?

Reply:
Yup, as serializable properties on the `Nullable<T>` type, we used to write those out per https://github.com/dotnet/runtime/issues/30843.